tremor-script/src/ast/query/raw.rs

```diff
@@ -181,6 +239,274 @@ impl<'script> Upable<'script> for OperatorDeclRaw<'script> {
+
+                let ports = subquery_decl.from.iter().chain(subquery_decl.into.iter());
+                for port in ports {
+                    let stream_stmt = StmtRaw::Stream(StreamStmtRaw {
+                        // TTODO Use start/end from Port for better errors
+                        start: self.start,
+                        end: self.end,
+                        id: port.id.to_string(),
+                    });
+                    subquery_stmts.push(stream_stmt);
+                }
+
+                subquery_stmts.extend(subquery_decl.raw_stmts.clone());
+                let subq_module = &self.mangle_id(&self.id);
+                helper.module.push(subq_module.clone());
+
+                let decl_params = subquery_decl.params.clone();
+                let stmt_params = up_maybe_params(self.params.clone(), &mut helper)?;
+                let subq_args = self.get_args_map(&decl_params, stmt_params, helper)?;
+
+                for stmt in subquery_stmts {
+                    match stmt {
+                        StmtRaw::ModuleStmt(m) => {
+                            m.define(helper.reg, helper.aggr_reg, &mut vec![], &mut helper)?;
+                        }
+                        StmtRaw::SubqueryStmt(mut s) => {
+                            s.id = self.mangle_id(&s.id);
+                            s.params = SubqueryStmtRaw::inline_maybe_params(
+                                s.params,
+                                &subq_args,
+                                &mut helper,
+                            )?
+                            .into();
+                            s.inline(&mut helper, &mut query_stmts)?;
+                        }
+                        StmtRaw::Operator(mut o) => {
+                            o.id = self.mangle_id(&o.id);
+                            o.params = SubqueryStmtRaw::inline_maybe_params(
+                                o.params,
+                                &subq_args,
+                                &mut helper,
+                            )?
+                            .into();
+                            let mut o = o.up(helper)?;
+                            o.module.push(subq_module.clone());
+                            query_stmts.push(Stmt::Operator(o));
+                        }
+                        StmtRaw::Script(mut s) => {
+                            s.id = self.mangle_id(&s.id);
+                            s.params = SubqueryStmtRaw::inline_maybe_params(
+                                s.params,
+                                &subq_args,
+                                &mut helper,
+                            )?
+                            .into();
+                            let mut s = s.up(helper)?;
+                            s.module.push(subq_module.clone());
+                            query_stmts.push(Stmt::Script(s));
+                        }
+                        StmtRaw::Stream(mut s) => {
```

Comment:
I think we should check if streams defined here do not overlap with ports otherwise there is the potential for great confusion on the user side